packages/secure_storage/amplify_secure_storage_dart/lib/src/platforms/amplify_secure_storage_windows.dart

```diff
@@ -11,119 +11,69 @@
-  ///
-  /// Reference: https://docs.microsoft.com/en-us/windows/win32/api/wincred/ns-wincred-credentiala
-  String get _username => config.windowsOptions.targetNamePrefix != null
-      ? config.windowsOptions.targetNamePrefix!
-      : config.defaultNamespace;
+  late final FileKeyValueStore keyValueStore = () {
+    final directory = config.windowsOptions.storagePath ?? applicationDirectory;
+    if (directory != null) {
+      return FileKeyValueStore(fileName: _fileName, path: directory);
+    }
+    return FileKeyValueStore(
+      fileName: _fileName,
+      path: 'tmp',
+      fs: MemoryFileSystem(),
+    );
+  }();
 
-  /// The name of the credential.
+  /// The name of the file that will be used to store
+  /// encrypted data for this instance.
   ///
-  /// The TargetName and Type members uniquely identify the credential.
-  ///
-  /// Reference: https://docs.microsoft.com/en-us/windows/win32/api/wincred/ns-wincred-credentiala
-  String _getTargetName(String key) => '$_username/$key';
+  /// Example: "com.amplify.auth.secure_storage.json" for
+  /// a scope of "auth".
+  String get _fileName => '${config.defaultNamespace}.secure_storage.json';
 
   @override
-  void write({required String key, required String value}) {
-    return using((Arena arena) {
-      final utf8Value = utf8.encode(value) as Uint8List;
-      final blob = utf8Value.allocatePointerWith(arena);
-      final targetName = _getTargetName(key).toNativeUtf16(allocator: arena);
-      final userName = _username.toNativeUtf16(allocator: arena);
-      final credential = arena<CREDENTIAL>()
-        ..ref.Type = CRED_TYPE_GENERIC
-        ..ref.TargetName = targetName
-        ..ref.Persist = CRED_PERSIST_LOCAL_MACHINE
-        ..ref.UserName = userName
-        ..ref.CredentialBlob = blob
-        ..ref.CredentialBlobSize = utf8Value.length;
-      final result = CredWrite(credential, CRED_FLAG_DEFAULT);
-      if (result != TRUE) {
-        final errorCode = GetLastError();
-        if (errorCode != ERROR_SUCCESS) {
-          throw _getExceptionFromErrorCode(errorCode);
-        }
-      }
-    });
+  Future<void> write({required String key, required String value}) async {
+    final encrypted = encryptString(value);
+    await keyValueStore.writeKey(key: key, value: encrypted);
   }
 
   @override
-  String? read({required String key}) {
-    return using((Arena arena) {
-      final credential = arena<Pointer<CREDENTIAL>>();
-      final targetName = _getTargetName(key).toNativeUtf16(allocator: arena);
-      final result = CredRead(
-        targetName,
-        CRED_TYPE_GENERIC,
-        CRED_FLAG_DEFAULT,
-        credential,
+  Future<String?> read({required String key}) async {
+    final rawData = await keyValueStore.readKey(key: key);
+    if (rawData == null) {
+      return null;
+    }
+    final Uint8List encrypted;
+    try {
+      encrypted = Uint8List.fromList(List<int>.from(rawData as List));
```

Comment:
I think this makes two copies of the list (List.from, then Uint8List.fromList). I would do `Uint8List.fromList((rawData as List).cast());` to avoid that. Is the only failure because of the cast? It's probably more reasonable to do an `is` check if that's the case.

Reply:
Updated to use .cast() over List.from.

I kept the try/catch block. `rawData` could not be a list, or it could be a List that contains data other than integers. Either of those would be problematic. I could do an `is` check to confirm it is a list, but I would still need to cast to get it to List<int> rather than List<dynamic>.